.dotnet/src/Custom/Batch/BatchOperationToken.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.ClientModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text.Json;
+
+#nullable enable
+
+namespace OpenAI.Batch;
+
+internal class BatchOperationToken : ContinuationToken
+{
+    public BatchOperationToken(string batchId)
+    {
+        BatchId = batchId;
+    }
+
+    public string BatchId { get; }
+
+    public override BinaryData ToBytes()
+    {
+        using MemoryStream stream = new();
+        using Utf8JsonWriter writer = new(stream);
+        writer.WriteStartObject();
+
+        writer.WriteString("batchId", BatchId);
+
+        writer.WriteEndObject();
+
+        writer.Flush();
+        stream.Position = 0;
+
+        return BinaryData.FromStream(stream);
+    }
+
+    public static BatchOperationToken FromToken(ContinuationToken continuationToken)
+    {
+        if (continuationToken is BatchOperationToken token)
+        {
+            return token;
+        }
+
+        BinaryData data = continuationToken.ToBytes();
+
+        if (data.ToMemory().Length == 0)
+        {
+            throw new ArgumentException("Failed to create BatchOperationToken from provided continuationToken.", nameof(continuationToken));
+        }
+
+        Utf8JsonReader reader = new(data);
```

Comment:
yeah, it would be really simpler if we had some JSON Path APIs, e.g.

```csharp
string batchId = JsonPath.Read<string>(data, "batchId"u8);
```

Reply:
For these internal token serialization/deserialization routines, I was really just going fast and have expected to come back and fine-tune/optimize when we start to generate them from TypeSpec.  But we can look at that sooner than later if you prefer.  Would it be OK to follow up on the non-trivial pieces of this separately to (hopefully) unblock getting the public LRO APIs into the API surface for the arch board review?

Reply:
Saving this to come back to in a separate PR.